### DISCUSSION TITLE
Allow accessing hook-modified payload in action event

### DISCUSSION THREAD

**Discussion by samdze** [directus/directus#13188] [Feature Requests]
I subscribed to the 'user.create' hook using both an action and a filter.
In the filter, I add an additional field to the payload and return it. The user gets created correctly with the added field.

The issue is inside the action hook. The payload object doesn't contain the additional field I added during the filter function.

### To Reproduce

Here's my setup:

```js
export default defineHook(({ filter, action }, { services }) => {
	const { ItemsService } = services

	filter('users.create', async (input, meta, { database, schema, accountability }) => {
		const profilesService = new ItemsService('profiles', { database, schema, accountability })
		const resultId = await profilesService.createOne({ first_name: input.first_name, last_name: input.last_name, email: input.email })

		console.log('User is being created!', { input, meta, resultId })
		return { ...input, profile: resultId }
	})

	action('users.create', async (input, { database, schema, accountability }) => {
                // input.payload doesn't contain 'profile'!
		console.log('User created!', input)
	})
})
```

**Comment by rijkvanzanten** [comment]
This is currently working as intended. The action hook's `payload` context key contains the original payload as submitted by the user, not the final-processed payload after all hooks and other processing has fired. Lets consider this discussion the feature request to add a new context flag that includes the final-processed payload that's inserted to the database

**Reply by samdze** [reply]
Yes, I think it would be valuable to somehow include the processed payload inside the action.
Also, I have a related issue: in the action above, if I try to read the just created user:
```
action('users.create', async (input, { database, schema, accountability }) => {
	try {
		const usersService = new ItemsService('directus_users', { database, schema })
		const { profile: profileId } = await usersService.readOne(input.key, { database, schema })

		...
	} catch (e) {
		console.error(e)
	}
})
```
**sometimes** I get a ForbiddenException, probably because the user is not yet fully created or readable.
I have to do this read precisely because I have not received the updated payload from the filter above.
Right now I have no solutions.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
